Accept JSON policy object in cluster policy commands

copdips opened the pull request:
#554

copdips commented:
@pietern 
could you please review ?

pietern reviewed (changes requested):
Thanks for making the PR, @copdips. Apologies for the longer turn around time.

copdips commented on databricks_cli/cluster_policies/api.py:
```diff
@@ -20,18 +20,42 @@
+    def create_cluster_policy(self, policy):
+        ClusterPolicyApi.format_policy_for_api(policy)
+        return self.client.client.perform_query(
+            "POST", "/policies/clusters/create", data=policy
```
fixed

copdips replied:
fixed

copdips commented on databricks_cli/cluster_policies/api.py:
```diff
@@ -20,18 +20,42 @@
+    def format_policy_for_api(policy):
+        policy["definition"] = ClusterPolicyApi.convert_to_json_string(
+            policy["definition"]
+        )
```
Thx for the review, I merged the two functions into one.
BTW, it seems that the `definition` key is indeed always set as per this API doc, so I'm using `policy["definition"]` instead of `policy.get`.
* create: https://docs.databricks.com/dev-tools/api/latest/policies.html#operation/create-cluster-policy
* edit :https://docs.databricks.com/dev-tools/api/latest/policies.html#operation/edit-cluster-policy

pietern replied:
It has to be, but it might not be when user specified.

copdips replied:
If we use `policy.get("definition", None)`, and user forgets to add the key `definition`, he will get an error only after calling the Databricks API, it's a network consommation.

But if we use `policy["definition"]`, user will get an error locally right now.

IMHO, if the user forgets the key, both methods will return an error, but the later one is faster.

copdips replied:
@pietern 
Please correct me if my understanding is wrong.

copdips commented:
@pietern 
please recheck now

pietern commented:
@copdips Thanks for updating. Everything looks good to me. There's one failing test that alerts on function prototype changes compare to master and it alerts on the argument name being changes from `json` to `policy`. Could you revert that rename? That way, if anyone depends on these APIs and uses keyword args, their calls won't break.

copdips commented:
> @copdips Thanks for updating. Everything looks good to me. There's one failing test that alerts on function prototype changes compare to master and it alerts on the argument name being changes from `json` to `policy`. Could you revert that rename? That way, if anyone depends on these APIs and uses keyword args, their calls won't break.

Thx the review, I reverted the rename, and I also changed `import json` to `from json import dumps as json_dumps`, otherwise pylint won't be happy

pietern reviewed (approved):
Thank you, @copdips ! Looks great.
